flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/DataStreamJavaITCase.java

```diff
@@ -531,6 +531,8 @@ void testFromChangelogStreamUpsert() {
                         output(RowKind.INSERT, "alice", 1),
                         // --
                         input(RowKind.INSERT, "alice", 1), // no impact
+                        input(RowKind.UPDATE_AFTER, "alice", 1), // no impact
```

Comment:
Could we add a `ChangelogNormalizeSematicTests` similar to `SinkSemanticTests`? This location is actually meant for testing DataStream<->Table API interop. We should start advertising this new and very useful testing infra.

Reply:
Also added `ChangelogNormalizeSematicTests`